crates/noirc_frontend/src/hir/def_collector/dc_mod.rs

```diff
@@ -183,6 +188,38 @@ impl<'a> ModCollector<'a> {
         }
     }
 
+    /// Collect any type aliases definitions declared within the ast.
+    /// Returns a vector of errors if any type aliases were already defined.
+    fn collect_type_aliases(
+        &mut self,
+        context: &mut Context,
+        type_aliases: Vec<NoirTypeAlias>,
+        errors: &mut Vec<FileDiagnostic>,
+    ) {
+        for type_alias in type_aliases {
+            let name = type_alias.name.clone();
+
+            let ty_alias_id = TypeAliasId(self.def_collector.collected_type_aliases.len());
```

Comment:
Should expose a function here to return the next type alias id from the node interner. We shouldn't do it from the def collector since that is reset for each crate.

Reply:
No, we shouldn't have any Nodes in the Arena for type aliases. I'm saying to use https://github.com/noir-lang/noir/pull/2003/files/fc6a41babb7121503cdd45c4a38cee68802e92a7#diff-cd6207a6ecb11933dab462d031dde535850c9ad10773df265d7e38ddd20ccba9R64